services/agent/src/simcore_service_agent/_settings.py

```diff
@@ -0,0 +1,43 @@
+from typing import Final
+
+from models_library.basic_types import LogLevel
+from pydantic import Field, NonNegativeInt, validator
+from settings_library.base import BaseCustomSettings
+from settings_library.r_clone import S3Provider
+
+_MIN: Final[NonNegativeInt] = 60
+
+
+class ApplicationSettings(BaseCustomSettings):
+    LOGLEVEL: LogLevel = Field(LogLevel.INFO)
+
+    AGENT_VOLUMES_CLEANUP_S3_SECURE: bool = False
```

Comment:
Why dont' you use ``packages/settings-library/src/settings_library/s3.py`` as in

```python
class ApplicationSettings(BaseCustomSettings):
      AGENT_S3: S3Settings = Field(auto_default_from_env=True)
```

The advantage is that you reduce the configuration burden since **all** services of the stack needeing access to S3 can be configured at once with

```env
S3_ACCESS_KEY=12345678
S3_BUCKET_NAME=simcore
S3_ENDPOINT=172.17.0.1:9001
S3_SECRET_KEY=12345678
S3_SECURE=0
```

If for any reason we want to point to different settings, we can then do it by setting ``AGENT_S3={S3_... }`` explicitly

Reply:
This exact example is what I want to avoid. I want the env vars to be different and not to be constraint to starting with `S3_`.
Let's take as an example `S3_SECURE`, I'd like for it to be `AGENT_VOLUMES_CLEANUP_S3_SECURE`. From my understanding this cannot be achieved with the above.

Reply:
The data from this backup sits in a separate destination from the one used in simcore, which could also be outside the current deployment, not just a bucket. It's configuration needs to be configured separately

Reply:
Yes you can achieve that by setting the env variable with a json  ``AGENT_S3="{S3_SECURE=False, ... }"``

Look at this example here 
https://github.com/ITISFoundation/osparc-simcore/blob/b2c873a1a7efd172ee4c9eeb38a7c8acaf9ac8c9/packages/settings-library/tests/test_base_w_postgres.py#L198


I mean, you can leave it as is now but PLEASE make sure you understand this concept, otherwise ask. We have gone several times through this :-(

Reply:
Think I did not really manage to convey my issue.

If we take a look at `S3Settings` it looks something like this:

```python
class S3Settings(BaseCustomSettings):
    S3_SECURE: bool = False
    S3_ENDPOINT: str
    S3_ACCESS_KEY: str
    S3_SECRET_KEY: str
    S3_ACCESS_TOKEN: Optional[str] = None
    S3_BUCKET_NAME: str
    S3_REGION: str = "us-east-1"
```

So let's say I use the following in a model:

```python
class SomeSettings(BaseCustomSettings):
    R_CLONE_S3: S3Settings = Field(auto_default_from_env=True)
```

Inside the instance of `SomeSettings` I can access the following `some_settings.R_CLONE_S3.S3_SECURE` and I am fine with that.

The settings above are capture from the following env vars:

```
S3_SECURE
S3_ENDPOINT
S3_ACCESS_KEY
S3_SECRET_KEY
S3_ACCESS_TOKEN
S3_BUCKET_NAME
S3_REGION
```
---

Now let's say I want to have a different configuration for s3. I'd like this configuration to be captured form these env vars. Note they look like the above but have the `ALTERNATIVE_` prefix.

```
ALTERNATIVE_S3_SECURE
ALTERNATIVE_S3_ENDPOINT
ALTERNATIVE_S3_ACCESS_KEY
ALTERNATIVE_S3_SECRET_KEY
ALTERNATIVE_S3_ACCESS_TOKEN
ALTERNATIVE_S3_BUCKET_NAME
ALTERNATIVE_S3_REGION
```

Reply:
I understand. What I mean is that there are two options for the "alternative" env vars
a) you create seven of them as ``ALTERNATIVE_`` that expect a value each
b) you create a single one ``ALTERNATIVE`` that expects  **a json** with the entire configuration

in the option b) you have in addition the possibility to capture ``S3_*``-like env vars if one day 

Your choice (a) works for your purpose so go ahead. I just wanted to make sure that you understand that we can also solve this problem with option b). And that b) is the recommended solution when we want to reuse the "default" s3 configuration.